content/blog/improved-refresh-destroy-experience/index.md

```diff
@@ -0,0 +1,237 @@
+---
+title: "Improved refresh and destroy experience for Pulumi IaC"
+date: 2025-04-02
+meta_desc: "Pulumi now runs your code before refresh and destroy operations, enabling dynamic credential updates and complex workflows"
+meta_image: meta.png
+authors:
+    - will-jones
+tags:
+    - features
+    - releases
+    - iac
+---
+
+Pulumi enables teams to manage their infrastructure using the programming languages and tools they are already familiar with, supporting use cases such as more complex authentication workflows, dynamically configured resources, and more. Loops, conditionals, environment variables, external calls.
+
+We're excited to announce an improvement to the `pulumi refresh` and `pulumi destroy` commands, a flag for `--run program`! This new feature makes Pulumi even more powerful for teams with complex infrastructure workflows.
+
+This enhancement is particularly valuable for teams working with short-lived credentials, dynamic resources, or any workflow where your code needs to run to establish the right context. Whether you're using OIDC-based authentication, dynamically fetching credentials from a secret manager, or working with dynamic providers that reference your Pulumi SDK, the `--run-program` flag ensures your infrastructure operations have the context they need to succeed.
```

Comment:
```suggestion
This enhancement is particularly valuable for teams working with short-lived credentials, dynamic resources, or any workflow where your code needs to run to establish the right context. Whether you're using OIDC-based authentication, or dynamically fetching credentials from a secret manager, the `--run-program` flag ensures your infrastructure operations have the context they need to succeed.
```

Reply:
Maybe worth a comment somewhere that this unblocks fixing things for dynamic providers as well.